mesonbuild/options.py

```diff
@@ -192,6 +195,25 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return f'OptionKey({self.name!r}, {self.subproject!r}, {self.machine!r})'
 
+    @classmethod
+    def factory(cls, name: str, subproject: str = '', machine: MachineChoice = MachineChoice.HOST) -> OptionKey:
```

Comment:
I haven't tried to implement it, but I think you can obtain the same effect without forcing the users to call the `.factory()` class method by implementing `__new__()` instead.